### PR TITLE
feat(curriculum): add css-flexbox-quiz

### LINKED ISSUE #56187: [HACKTOBERFEST 2024] - CSS Flexbox Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the CSS Flexbox quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-css-flexbox`.

**All quiz questions need to relate to one of these topics:**

- what is flexbox
- common flex properties
  - flex-wrap
  - justify-content
  - align-items

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by henrysmile2071**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56187 

1. Ran locally passing: ```FCC_CHALLENGE_ID=66ed8fe7f45ce3ece4053eb2 pnpm run test:curriculum```

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 1]
```diff
@@ -17,439 +17,524 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --distractors--
 
-Placeholder distractor 1
+```css
+
+div { 
+  flex: 1; 
+}
+
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:51]:
```suggestion
div { 
  flex: 1; 
}
```
No need for additional blank lines here. It applies to other code blocks as well.

Comment (henrysmile2071) [reply curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md]: Ok, I've removed the additional blank lines for all the code blocks

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 2]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+An one-dimensional model for layout 
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What css property is set to enable the flexbox layout for the <div> element?
```
Comment (a2937) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:42]: I'm fairly sure that we want HTML elements like div to have backticks around them like this: `div`.

Comment (henrysmile2071) [reply curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md]: Oh, thanks for the catch, I have added the backticks.

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 3]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --distractors--
 
-Placeholder distractor 1
+Items are packed flush to each other toward the start edge of the container
 
 ---
 
-Placeholder distractor 2
+Items are aligned such that their flex container baselines align
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:333]:
```suggestion
Items are aligned such that their flex container baselines align.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 4]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 3
+Items are packed flush to each other toward the end edge of the container
 
 #### --answer--
 
-Placeholder answer
+Auto-size items to be equally large to fill the container
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:341]:
```suggestion
Auto-size items to be equally large to fill the container.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 5]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides are flush againt the main axis edge. Which value in `justify-content` matches this description?
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:347]:
```suggestion
Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides are flush againt the main axis edge.

Which value in `justify-content` matches this description?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 6]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+`space-between`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items and against the main axis edge are the same. Which value in `justify-content` matches this description?
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:369]:
```suggestion
Items are evenly distributed along the main axis. The spacing between each pair of adjacent items and against the main axis edge are the same.

Which value in `justify-content` matches this description?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 7]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+`space-evenly`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides have half the spacing against the main axis edge. Which value in `justify-content` matches this description?
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:391]:
```suggestion
Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides have half the spacing against the main axis edge.

Which value in `justify-content` matches this description?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 8]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+`space-around`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are flush against the cross-axis start. Which value in `align-items` matches this description?
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:413]:
```suggestion
Items are flush against the cross-axis start.

Which value in `align-items` matches this description?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 9]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+`flex-start`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are flush against the cross-axis end. Which value in `align-items` matches this description?
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:435]:
```suggestion
Items are flush against the cross-axis end.

Which value in `align-items` matches this description?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 10]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+`flex-end`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are aligned along the main axis from left to right. Which value in `flex-direction` matches this description?
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:457]:
```suggestion
Items are aligned along the main axis from left to right.

Which value in `flex-direction` matches this description?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 11]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --text--
 
-Placeholder question
+Which of the following use case is not suitable for flexbox?
 
 #### --distractors--
 
-Placeholder distractor 1
+Centering an item in a container
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:505]:
```suggestion
Centering an item in a container.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 12]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 2
+Vertical alignment of items in a row
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:509]:
```suggestion
Vertical alignment of items in a row.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 13]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 ---
 
-Placeholder distractor 2
+Vertical alignment of items in a row 
 
 ---
 
-Placeholder distractor 3
+Dynamically adjusting distribution and alignment of items for different viewports
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:513]:
```suggestion
Dynamically adjusting distribution and alignment of items for different viewports.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 14]
```diff
@@ -17,439 +17,502 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
 #### --answer--
 
-Placeholder answer
+Fixed sizing and space between items like a grid
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:517]:
```suggestion
Fixed sizing and space between items like a grid.
```

**Comment by ilenia-magoni** [comment]
please make a batch from the Edited files tab, do not accept each suggestion as a commit individually

**Comment by henrysmile2071** [comment]
Oh, sorry I wasn't familar with the batch feature, I'll use git squash then.

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 15]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
+`flex-direction`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which CSS matches the following description?
+
+A container where elements flow from left to right, changes line when it exceeds containter width and new lines go on top of the previous line.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:148]:
```suggestion
What CSS properties would you use to make a `.container` element's items flow from left to right, change lines when an item exceeds container width, and have new items go on top of the previous line?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 16]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
+`row`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which CSS matches the following description?
+
+Center an element within a container horizontally and vertically.
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:240]:
```suggestion
What CSS properties would you use to center an element within a flex container horizontally, and vertically?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 17]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
-Placeholder answer
+Auto-size items to be equally large to fill the container.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides are flush againt the main axis edge. 
+Which value in `justify-content` matches this description?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:350]:
```suggestion
Which value for `justify-content` will evenly distribute items along the main axis, have the same spacing between each pair of adjacent items, and have items whose sides are flush againt the main axis edge?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 18]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
-Placeholder answer
+`space-between`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items and against the main axis edge are the same. 
+Which value in `justify-content` matches this description?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:373]:
```suggestion
Which value for `justify-content` will evenly distribute items along the main axis, and have the same spacing between each pair of adjacent items and against the main axis edge?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 19]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
-Placeholder answer
+`space-evenly`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are evenly distributed along the main axis. The spacing between each pair of adjacent items is the same. The items on the sides have half the spacing against the main axis edge. 
+Which value in `justify-content` matches this description?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:396]:
```suggestion
Which value for `justify-content` will evenly distribute items along the main axis, have the same spacing between each pair of adjacent items, and have items on the sides with half the spacing against the main axis edge?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 20]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
+`space-around`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are flush against the cross-axis start.
+
+Which value in `align-items` matches this description?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:420]:
```suggestion
Which value for `align-items` will make it so items are flush against the cross-axis start?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 21]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
+`flex-start`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are flush against the cross-axis end.
+
+Which value in `align-items` matches this description?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:444]:
```suggestion
Which value for `align-items` will make it so items are flush against the cross-axis end?
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 22]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
+`flex-end`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are aligned along the main axis from left to right.
+
+Which value in `flex-direction` matches this description?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:468]:
```suggestion
Which value for `flex-direction` will make it so items are aligned along the main axis from left to right.
```

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 23]
```diff
@@ -17,439 +17,514 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
-Placeholder answer
+`row`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Items are aligned along the cross axis from top to bottom. 
+Which value in `flex-direction` matches this description?
```
Comment (moT01) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:491]:
```suggestion
Which value for `flex-direction` will make it so items are aligned along the cross axis from top to bottom.
```

**Review by moT01 — COMMENTED** [review]
This looks good @henrysmile2071 🎉 I like these questions. I would be fine with this as-is, but I made some suggestions to turn some of the questions into single sentences. It's really just a personal preference. I think they're a little more concise and look a little better. Feel free to let me know if you have any issues with any of them, or just want to leave them as they are - some of them do get a little lengthy.

**Comment by henrysmile2071** [comment]
> This looks good @henrysmile2071 🎉 I like these questions. I would be fine with this as-is, but I made some suggestions to turn some of the questions into single sentences. It's really just a personal preference. I think they're a little more concise and look a little better. Feel free to let me know if you have any issues with any of them, or just want to leave them as they are - some of them do get a little lengthy.

@moT01 Thanks, they look much better😁 I will change them as you suggested.

**Review by moT01 — APPROVED** [review]
LGTM @henrysmile2071 🎉 Thanks for making those adjustments 👍

File: curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md [hunk 24]
```diff
@@ -17,439 +17,500 @@ To pass the quiz, you must correctly answer at least 17 of the 20 of the questio
...
 
-Placeholder answer
+`row`
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What does the `stetch` value in `align-items` do?
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/quiz-css-flexbox/66ed8fe7f45ce3ece4053eb2.md:323]:
```suggestion
What does the `stretch` value in `align-items` do?
```

**Review by Dario-DC — COMMENTED** [review]
Found a typo. Other than that it looks good to me.

**Review by Dario-DC — APPROVED** [review]
Hi @henrysmile2071 . Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**